rerun_cpp/CMakeLists.txt

```diff
@@ -97,33 +90,133 @@ if("${RERUN_C_LIB}" STREQUAL "")
+        elseif(WIN32)
+            set(ARROW_LIBRARY_FILE ${ARROW_DOWNLOAD_PATH}/lib/arrow_static.lib)
+        else()
+            message(FATAL_ERROR "Unsupported platform.")
+        endif()
+    endif()
+
+    # Enable multithreaded compiling of Arrow on MSVC.
+    if(MSVC)
+        set(DARROW_CXXFLAGS "/MP")
+    else()
+        set(DARROW_CXXFLAGS "")
+    endif()
+
+    ExternalProject_Add(
+        arrow_cpp
+        PREFIX ${ARROW_DOWNLOAD_PATH}
+        GIT_REPOSITORY https://github.com/apache/arrow.git
+        GIT_TAG apache-arrow-10.0.1
+        GIT_SHALLOW true
+        CMAKE_ARGS
+        --preset ninja-debug-minimal
+        -DARROW_IPC=ON
+        -DARROW_BUILD_SHARED=${ARROW_BUILD_SHARED}
+        -DARROW_BUILD_STATIC=${ARROW_BUILD_STATIC}
+        -DCMAKE_INSTALL_PREFIX=${ARROW_DOWNLOAD_PATH}
+        -DARROW_USE_ASAN=OFF
+        -DARROW_USE_TSAN=OFF
+        -DARROW_USE_UBSAN=OFF
+        -Dxsimd_SOURCE=BUNDLED
+        -DBOOST_SOURCE=BUNDLED
+        -DARROW_BOOST_USE_SHARED=OFF
+        -DARROW_JEMALLOC=OFF
+        -DARROW_CXXFLAGS=${DARROW_CXXFLAGS}
+        SOURCE_SUBDIR cpp
+        BUILD_BYPRODUCTS ${ARROW_LIBRARY_FILE}
+    )
+
+    # arrow_cpp target is not a library. Assemble one from it.
+    if(RERUN_ARROW_LINK_SHARED)
+        add_library(RerunArrowTarget SHARED IMPORTED)
 
-find_package(Arrow REQUIRED)
+        # For windows we need to know both the dll AND the import library.
+        if(WIN32)
+            set_target_properties(RerunArrowTarget PROPERTIES IMPORTED_IMPLIB ${ARROW_DOWNLOAD_PATH}/lib/arrow.lib)
+        endif()
+    else()
+        add_library(RerunArrowTarget STATIC IMPORTED)
+        target_compile_definitions(RerunArrowTarget INTERFACE ARROW_STATIC)
+    endif()
 
-if(RERUN_ARROW_LINK_SHARED)
-    message(STATUS "Arrow SO version: ${ARROW_FULL_SO_VERSION}")
-    target_link_libraries(rerun_sdk PRIVATE Arrow::arrow_shared)
+    add_dependencies(RerunArrowTarget arrow_cpp)
+    set_target_properties(RerunArrowTarget PROPERTIES
+        IMPORTED_LOCATION ${ARROW_LIBRARY_FILE}
+        INTERFACE_INCLUDE_DIRECTORIES ${ARROW_DOWNLOAD_PATH}/include
+    )
+
+    # Hack to propagate INTERFACE_INCLUDE_DIRECTORIES.
+    # via https://stackoverflow.com/a/47358004
+    file(MAKE_DIRECTORY ${ARROW_DOWNLOAD_PATH}/include)
 else()
-    message(STATUS "Arrow version: ${ARROW_VERSION}")
+    find_package(Arrow REQUIRED)
 
-    if(WIN32)
-        message(WARNING "As of writing static linking with libarrow is broken on Windows. See https://github.com/apache/arrow/issues/33145")
+    if(RERUN_ARROW_LINK_SHARED)
+        message(STATUS "Arrow SO version: ${ARROW_FULL_SO_VERSION}")
+        add_library(RerunArrowTarget ALIAS Arrow::arrow_shared)
+    else()
+        message(STATUS "Arrow version: ${ARROW_VERSION}")
+        add_library(RerunArrowTarget ALIAS Arrow::arrow_static)
     endif()
 
-    target_link_libraries(rerun_sdk PRIVATE Arrow::arrow_static)
+    set(RERUN_ARROW_TARGET Arrow::arrow_shared)
```

Comment:
Shouldn't this me inside the `if` with
```
set(RERUN_ARROW_TARGET Arrow::arrow_static)
```
in the `else`?

Reply:
Or maybe this isn't used at all?

Reply:
oh. yeah that's a broken left over